app/controllers/glossary_terms_controller.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+# create and edit Glossary terms
+class GlossaryTermsController < ApplicationController
+  before_action :login_required, except: [
+    :index,
+    :show,
+    :show_past_glossary_term,
+    :show_glossary_term # aliased
+  ]
+
+  def show
+    store_location
+    @glossary_term = GlossaryTerm.find(params[:id].to_s)
+    @canonical_url = "#{MO.http_domain}/glossary/#{@glossary_term.id}"
+    @layout = calc_layout_params
+    @objects = @glossary_term.images
+  end
+
+  alias_method :show_glossary_term, :show
+
+  # Show past version of GlossaryTerm.
+  # Accessible only from show_glossary_term page.
+  def show_past_glossary_term # :prefetch: :norobots:
+    pass_query_params
+    store_location
+    if @glossary_term = find_or_goto_index(GlossaryTerm, params[:id].to_s)
+      if params[:version]
+        @glossary_term.revert_to(params[:version].to_i)
+      else
+        flash_error(:show_past_location_no_version.t)
+        redirect_to glossary_term_path(@glossary_term.id)
+      end
+    end
+  end
+
+  def index
+    store_location
+    @glossary_terms = GlossaryTerm.all.order(:name)
+  end
+
+  def new
+    @copyright_holder = @user.name
+    @copyright_year = Time.now.utc.year
+    @upload_license_id = @user.license_id
+    @licenses = License.current_names_and_ids(@user.license)
+  end
+
+  alias_method :create_glossary_term, :new
+  alias_method :glossary_term_get, :new
+
+  def create
+    @glossary_term = \
+      GlossaryTerm.new(user: @user, name: params[:glossary_term][:name],
+                       description: params[:glossary_term][:description])
+    @glossary_term.add_image(process_image(image_args))
+    @glossary_term.save
+    redirect_to glossary_term_path(@glossary_term.id)
+  end
+
+  alias_method :glossary_term_post, :create
+
+  def edit
+    # Expand to any MO user,
+    # but make them owned and editable only by that user or an admin
+    @glossary_term = GlossaryTerm.find(params[:id].to_s)
+  end
+
+  alias_method :edit_glossary_term, :edit
+
+  def update
+    @glossary_term = GlossaryTerm.find(params[:id].to_s)
+    @glossary_term.attributes = params[:glossary_term].
+                                  permit(:name, :description)
+    @glossary_term.user = @user
+    @glossary_term.save
+    redirect_to glossary_term_path(@glossary_term.id)
+  end
+
+  # no alias needed
+
+  ##############################################################################
```

Comment:
I plan on adding a ':destroy` action:
- Available only to admins
- Should be linked -only in admin mode -- from :show and :index views